### ISSUE TITLE
[Linter][LocationMustHaveXmsMutability] Reports broken path

### ISSUE THREAD

**Issue by vishrutshah** [Azure/azure-openapi-validator#97]
Currently, we have two messages for one error but with a different format in CI. Example: https://travis-ci.com/Azure/azure-rest-api-specs-pr/jobs/86104329#L693

One is 
```json5
{
  "type": "Warning",
  "code": "LocationMustHaveXmsMutability",
  "message": "Property 'location' must have '\"x-ms-mutability\":[\"read\", \"create\"]' extension defined. Resource Model: 'LocationServicesAccount'",
  "id": "S4002",
  "validationCategory": "SDKViolation",
  "providerNamespace": null,
  "resourceType": null,
  "sources": [
    "file:///home/travis/build/Azure/azure-rest-api-specs-pr/specification/locationservices/resource-manager/Microsoft.LocationServices/2017-01-01-preview/locationservices.json:469:2 ($.definitions)"
  ]
}
```

another is
```json5
{
  "Channel": "warning",
  "Text": "Could not find the exact path [\"definitions\",\"properties\",\"location\"] for {\"type\":\"Warning\",\"code\":\"LocationMustHaveXmsMutability\",\"message\":\"Property 'location' must have '\\\"x-ms-mutability\\\":[\\\"read\\\", \\\"create\\\"]' extension defined. Resource Model: 'LocationServicesAccount'\",\"id\":\"S4002\",\"validationCategory\":\"SDKViolation\",\"providerNamespace\":null,\"resourceType\":null}"
}
```

The second one tells us that broken path is being reported by this linter rule. Example: `[\"definitions\",\"properties\",\"location\"]`. There is something missing between `definitions` & `properties`

cc:
@veronicagg @dsgouda @olydis  

https://github.com/Azure/autorest/pull/2590#issuecomment-329881024

**Comment by veronicagg** [comment]
@vishrutshah @dsgouda is this already addressed? can it be closed?

**Comment by dsgouda** [comment]
I'm not sure if this was addressed.

**Comment by olydis** [comment]
yes, this is printed as debug or verbose now (not sure which), and will hence not appear in stdout/stderr unless you ask for it

**Comment by dsgouda** [comment]
Wait, what @olydis  meant was the reported broken paths are now being suppressed, the issue still stands and we need to check if the correct paths are being returned by the validation rule.

**Comment by veronicagg** [comment]
ok, reopening then.
